qiskit/primitives/fidelity/base_fidelity.py

```diff
@@ -0,0 +1,122 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Base fidelity primitive
+"""
+
+from __future__ import annotations
+from abc import ABC, abstractmethod
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import ParameterVector
+
+
+class BaseFidelity(ABC):
+    """
+    Defines the interface to calculate fidelities.
+    """
+
+    def __init__(
+        self,
+        left_circuit: QuantumCircuit | None = None,
+        right_circuit: QuantumCircuit | None = None,
+    ) -> None:
+        r"""Initializes the class to evaluate the fidelities defined as
+
+            :math:`|\langle\psi(x)|\phi(y)\rangle|^2`,
+
+        where :math:`x` and :math:`y` are optional parametrizations of the
+        states :math:`\psi` and :math:`\phi` prepared by the circuits
+        ``left_circuit`` and ``right_circuit``, respectively.
+
+        Args:
+            left_circuit: (Parametrized) quantum circuit preparing :math:`|\psi\rangle`.
+            right_circuit: (Parametrized) quantum circuit preparing :math:`|\phi\rangle`.
+
+        Raises:
+            ValueError: ``left_circuit`` and ``right_circuit`` don't have the same number of qubits.
+        """
+
+        if left_circuit is None or right_circuit is None:
```

Comment:
From the docs and signature it looks like I can set the circuits independently, but the code only works if I set both at the same time -- do we want to allow independent setting?

Reply:
For the estimator one can pass a list of circuits in. I am not sure it makes sense for both left and right to be lists here but maybe allowing one to be a list so you can do multiple fidelities?

Reply:
We are currently looking into passing lists and how this can enable batching in fidelity-based algorithms.